dsgrid/registry/dataset_registry_manager.py

```diff
@@ -127,37 +143,123 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
+        if len(missing_dimensions) != 0:
             raise DSGInvalidDataset(
-                f"load_data timestamps do not match expected times. mismatch={mismatch}"
+                f"load_data_one_table is missing dimensions: {missing_dimensions}. If these are trivial dimensions, make sure to specify them in the Dataset Config."
             )
 
-        timestamps_by_id = (
-            load_data.select("timestamp", "id")
-            .groupby("id")
-            .agg(F.countDistinct("timestamp").alias("distinct_timestamps"))
-        )
-        for row in timestamps_by_id.collect():
-            if row.distinct_timestamps != len(expected_timestamps):
+        dimension_types = [
+            d for d in dimension_types if d != DimensionType.TIME
+        ]  # remove time type
+        for dimension_type in dimension_types:
+            name = dimension_type.value
+            dimension = config.get_dimension(dimension_type)
+            dim_records = dimension.get_unique_ids()
+            if dimension_type == DimensionType.METRIC:
+                data_records = set(metric_cols)
+            else:
+                data_records = get_unique_values(load_data_one_table, name)
+            if dim_records != data_records:
+                logger.error(
+                    "Mismatch in load_data_one_table records. dimension=%s mismatched=%s",
+                    name,
+                    data_records.symmetric_difference(dim_records),
+                )
+                raise DSGInvalidDataset(
+                    f"load_data_one_table records do not match dimension records for {name}"
+                )
+
+    def _check_dataset_time_consistency(self, config: DatasetConfig, load_data):
+        """
+        Common time check.
+        Only check time if it's not ANNUAL time type
+        """
+        time_dim = config.get_dimension(DimensionType.TIME)
+
+        if time_dim.model.time_type != TimeDimensionType.ANNUAL:
+            tz = time_dim.get_tzinfo()
+            time_ranges = time_dim.get_time_ranges()
+            assert len(time_ranges) == 1, len(time_ranges)
+            time_range = time_ranges[0]
+            # TODO: need to support validation of multiple time ranges: DSGRID-173
+
+            expected_timestamps = time_dim.list_time_range(time_range)
+            actual_timestamps = [
+                x.timestamp
+                for x in load_data.select("timestamp").distinct().sort("timestamp").collect()
+            ]
+            try:
+                actual_timestamps = [x.astimezone().astimezone(tz) for x in actual_timestamps]
+            except AttributeError:
+                if tz == None:
+                    actual_timestamps = [
+                        pd.Timestamp(str(x)).to_pydatetime() for x in actual_timestamps
+                    ]  # tz=tz does not work here when tz=None
+                else:
+                    actual_timestamps = [
```

Comment:
If I'm reading this correctly, in case 1 we'll have python datetime objects and in case 2 we'll have pandas.Timestamp. Will that be problematic later?

Reply:
Both are date time objects (i.e. converted from pandas using `.topydatetime()`), another way to get date time objects would be directly using `datetime.datetime(year=year, month=1, day=1, tzinfo=tz)` but that's much lengthier. Since we do use pd.Timestamp in the time class (https://github.com/dsgrid/dsgrid/blob/8cc304b031edb57ca3139b78ebdf52d1b3145e62/dsgrid/config/dimension_config.py#L101), I thought it would be okay here.

Reply:
Sorry, didn't read closely enough.